lib/resourceapply/apps.go

```diff
@@ -29,13 +29,19 @@ func ApplyDeploymentv1(ctx context.Context, client appsclientv1.DeploymentsGette
 		return nil, false, nil
 	}
 
+	var original appsv1.Deployment
+	existing.DeepCopyInto(&original)
 	modified := pointer.BoolPtr(false)
 	resourcemerge.EnsureDeployment(modified, existing, *required)
 	if !*modified {
 		return existing, false, nil
 	}
 	if reconciling {
-		klog.V(2).Infof("Updating Deployment %s/%s due to diff: %v", required.Namespace, required.Name, cmp.Diff(existing, required))
+		if diff := cmp.Diff(&original, existing); diff != "" {
```

Comment:
Why are we changing this and doing another diff here? This would differ from how it's done everywhere else (which of course doesn't mean it's wrong but do need a good reason for the change).

Reply:
TL;DR I'm convinced that the existing diff is incorrect and misleading.

I've tried to explain in the commit message https://github.com/openshift/cluster-version-operator/pull/851/commits/6b6412d64d025b883e88fc58a470b4fe70f2ebaf and in more details on [Slack](https://coreos.slack.com/archives/CJ1J9C3V4/p1666285479676509?thread_ts=1666284447.320419&cid=CJ1J9C3V4) when Trevor asked.

Sorry for not making the change description more prominent. Good point about "everywhere else" - I haven't realized we show the diff even for other resources. Can you check my explanation above and if this fix is correct, I can either make a followup PR to make all diffs like this, or I can extract this change to a separate PR entirely - what do you prefer?